nvme-print-json.c

```diff
@@ -11,14 +11,68 @@
 #include "common.h"
 
 #define ERROR_MSG_LEN 100
-#define STR_LEN 40
+#define STR_LEN 100
 #define BUF_LEN 320
 #define VAL_LEN 4096
 #define BYTE_TO_BIT(byte) ((byte) * 8)
+#define POWER_OF_TWO(exponent) (1 << (exponent))
+#define MS500_TO_SEC(time) ((time) / 2)
+
+#define array_add_obj(o, k) json_array_add_value_object(o, k);
+#define obj_add_str(o, k, v) json_object_add_value_string(o, k, v)
+#define root_add_array(k, v) json_object_add_value_array(root, k, v)
```

Comment:
I was wondering if wouldn't make sense to just use `r` instead of `root` as variable name in the functions and the use `obj_add_str(r, ..., ...)` instead introducing a wrapper macro just for this. What do you think?

As this series is pretty big and I don't want to make you suffer too much by updating every patch again, you can  just add a cleanup patch add the end of the series, that is fine with me.